src/main/java/blusunrize/immersiveengineering/client/gui/CircuitTableScreen.java

```diff
@@ -0,0 +1,256 @@
+						));
+				}
+			}
+		}
+	}
+
+	@Override
+	public void render(MatrixStack transform, int mx, int my, float partial)
+	{
+		super.render(transform, mx, my, partial);
+
+		List<ITextComponent> tooltip = new ArrayList<>();
+
+		if(this.hoveredSlot!=null&&this.hoveredSlot.slotNumber < SLOT_TYPES.length&&!this.hoveredSlot.getHasStack())
+		{
+			int slotNum = this.hoveredSlot.slotNumber;
+			tooltip.add(TextUtils.applyFormat(
+					new TranslationTextComponent(Lib.DESC_INFO+"circuit_table.slot."+SLOT_TYPES[slotNum]),
+					TextFormatting.GRAY
+			));
+		}
+		if(isMouseIn(mx, my, 217, 16, 7, 46))
+			tooltip.add(new StringTextComponent(tile.getEnergyStored(null)+"/"+tile.getMaxEnergyStored(null)+" IF"));
+
+		if(isMouseIn(mx, my, 52, 7, 100, 70)&&this.playerInventory.getItemStack().getItem() instanceof LogicCircuitBoardItem)
+			tooltip.add(TextUtils.applyFormat(
+					new TranslationTextComponent(Lib.DESC_INFO+"circuit_table.copy"),
+					TextFormatting.GRAY
+			));
+
+		for(GuiButtonState<LogicCircuitRegister> input : this.inputButtons)
+			if(input.isHovered())
+				tooltip.add(TextUtils.applyFormat(input.getState().getDescription(), TextFormatting.GRAY));
+		if(this.outputButton.isHovered())
+			tooltip.add(TextUtils.applyFormat(outputButton.getState().getDescription(), TextFormatting.GRAY));
+
+		if(!tooltip.isEmpty())
+			GuiUtils.drawHoveringText(transform, tooltip, mx, my, width, height, -1, font);
+	}
+
+	@Override
+	protected void drawGuiContainerForegroundLayer(MatrixStack transform, int mouseX, int mouseY)
+	{
+		drawCenteredString(transform, this.font, "Operator:", 76, 4, DyeColor.LIGHT_GRAY.getColorValue());
+		drawCenteredString(transform, this.font, "Inputs:", 130, 8, DyeColor.LIGHT_GRAY.getColorValue());
+		drawCenteredString(transform, this.font, "Outputs:", 130, 42, DyeColor.LIGHT_GRAY.getColorValue());
+
+		for(int i = 0; i < SLOT_TYPES.length; i++)
+		{
+			int amount = 0;
+			DyeColor col = DyeColor.LIGHT_GRAY;
+			if(this.instruction.get().isPresent())
+			{
+				amount = CircuitTableTileEntity.getIngredientAmount(this.instruction.get().get(), i);
+				if(this.container.inventorySlots.get(i).getStack().getCount() >= amount)
+					col = DyeColor.GREEN;
+				else
+					col = DyeColor.RED;
+			}
+			this.font.drawString(transform, "x "+amount, 30, 18+20*i, col.getColorValue());
+		}
+	}
+
+	@Override
+	protected void drawGuiContainerBackgroundLayer(MatrixStack transform, float f, int mx, int my)
+	{
+		RenderSystem.color3f(1.0F, 1.0F, 1.0F);
+		ClientUtils.bindTexture(TEXTURE);
+		this.blit(transform, guiLeft, guiTop, 0, 0, xSize, ySize);
+
+		int stored = (int)(46*(tile.getEnergyStored(null)/(float)tile.getMaxEnergyStored(null)));
+		fillGradient(transform, guiLeft+217, guiTop+16+(46-stored), guiLeft+224, guiTop+62, 0xffb51500, 0xff600b00);
+	}
+
+	@Override
+	public boolean keyPressed(int keyCode, int scanCode, int modifiers)
+	{
+		return super.keyPressed(keyCode, scanCode, modifiers);
+	}
```

Comment:
I think this can be removed now? Unless you're overriding it to change access modifiers or some other non-obvious reason, in which case there should be a comment IMO.